sim/common/vanilla/item_effects.go

```diff
@@ -2371,13 +2375,28 @@ func init() {
 			Name:     "Fiery Aura Proc",
 			Callback: core.CallbackOnSpellHitTaken,
 			Outcome:  core.OutcomeLanded,
-			ProcMask: core.ProcMaskMelee, // TODO: Unsure if this means melee attacks or all attacks
+			ProcMask: core.ProcMaskMelee,
 			Handler: func(sim *core.Simulation, spell *core.Spell, result *core.SpellResult) {
 				procSpell.Cast(sim, spell.Unit)
 			},
 		})
 	})
 
+	// https://www.wowhead.com/classic/item=228293/essence-of-the-pure-flame
+	// Equip: When struck in combat inflicts 100 Nature damage to the attacker.  Causes twice as much threat as damage dealt.
+	core.NewItemEffect(RazorbrambleLeathers, func(agent core.Agent) {
+		DamageShieldWithThreatMod(agent.GetCharacter(), 1213813, 100, 2, "Damage Shield Razorbramble Leathers")
+	})
+	core.NewItemEffect(RazorbrambleShoulderpads, func(agent core.Agent) {
+		DamageShieldWithThreatMod(agent.GetCharacter(), 1213816, 80, 2, "Damage Shield Razorbramble Shoulderpads")
+	})
+	core.NewItemEffect(RazorbrambleCowl, func(agent core.Agent) {
+		DamageShieldWithThreatMod(agent.GetCharacter(), 1213813, 100, 2, "Damage Shield Razorbramble Cowl")
+	})
+	core.NewItemEffect(LodestoneofRetaliation, func(agent core.Agent) {
+		DamageShieldWithThreatMod(agent.GetCharacter(), 1213816, 80, 2, "Damage Shield Lodestone of Retaliation")
+	})
```

Comment:
I didn't realize these were included here but I already implemented these in https://github.com/wowsims/sod/blob/master/sim/common/sod/item_effects/phase_6.go the other day while adding in all of the missing Thorns effects